### PR TITLE
Identified collections

### PR DESCRIPTION
**Pull request by micbakos-rdx**
* Used macro that replaces typealiases with structs for Personas, Accounts, FactorSources and AuthorizedDapps.
* Two macros are defined instead of using literal of true/false.
    *  `decl_can_be_empty_identified_array_of`: For empty collections
    * `decl_never_empty_identified_array_of`: For never empty collections
* FactorSources don't allow 0 elements
* `Default` trait is implemented only to `decl_can_be_empty_identified_array_of` macro.

### CODE REVIEW

File: src/profile/v100/factors/factor_source.rs [hunk 1]
```diff
@@ -43,6 +43,14 @@ impl BaseIsFactorSource for FactorSource {
...
 }
 
+impl Identifiable for FactorSource {
+    type ID = FactorSourceID;
+
+    fn id(&self) -> Self::ID {
+        self.factor_source_id()
+    }
+}
+
```
Comment (micbakos-rdx) [inline src/profile/v100/factors/factor_source.rs:53]: Moved in here

**Review by CyonAlexRDX — APPROVED** [review]
Excellent!

**Comment by micbakos-rdx** [comment]
@CyonAlexRDX Here is the unti test of uniffi converter + some common methods. https://github.com/radixdlt/sargon/pull/78/commits/ccc1b81535c0ff3ae45cc15c0c7236f204af4481

**Comment by CyonAlexRDX** [comment]
> @CyonAlexRDX Here is the unti test of uniffi converter + some common methods. https://github.com/radixdlt/sargon/pull/78/commits/ccc1b81535c0ff3ae45cc15c0c7236f204af4481

Nice! Looks good to be merged then?!

**Pull request merged**